File: FoodOn_backend/src/main/java/com/foodon/foodon/meal/domain/MealTimeType.java
package com.foodon.foodon.meal.domain;

import lombok.Getter;

@Getter
public enum MealTimeType {
    BREAKFAST("아침식사"),
    LUNCH("점심식사"),
    DINNER("저녁식사"),
    SNACK("간식");

    private final String description;

    MealTimeType(String description) {
        this.description = description;
    }
}

File: android-app/app/src/main/java/com/swallaby/foodon/domain/main/model/MealTimeType.kt
package com.swallaby.foodon.domain.main.model

import com.swallaby.foodon.domain.main.model.MealTimeType.values

enum class MealTimeType(val korean: String) {
    BREAKFAST("아침 식사"),
    LUNCH("점심 식사"),
    DINNER("저녁 식사"),
    SNACK("간식");

    companion object {
        fun fromKorean(korean: String): MealTimeType {
            return values().firstOrNull { it.korean == korean } ?: BREAKFAST
        }
    }
}